std/string.d

```diff
@@ -2491,6 +2509,11 @@ C[] stripLeft(C)(C[] str) @safe pure @nogc
            "hello world" ~ [lineSep]);
     assert(stripLeft([paraSep] ~ "hello world" ~ paraSep) ==
            "hello world" ~ [paraSep]);
+
+    import std.utf : byChar;
+    import std.array;
+    assert(stripLeft("     hello world     "w.byChar).array ==
```

Comment:
Instead of calling .array, you should be able to use .equal to verify equality and you won't have an allocation.